acceptance/openstack/imageservice/v2/images_test.go

```diff
@@ -78,3 +79,91 @@ func TestImagesCreateDestroyEmptyImage(t *testing.T) {
 
 	tools.PrintResource(t, image)
 }
+
+func TestImagesListByDate(t *testing.T) {
+	client, err := clients.NewImageServiceV2Client()
+	if err != nil {
+		t.Fatalf("Unable to create an image service client: %v", err)
+	}
+
+	date := time.Date(2014, 1, 1, 1, 1, 1, 0, time.UTC)
+	listOpts := images.ListOpts{
+		Limit: 1,
+		CreatedAt: &images.ImageDateQuery{
+			Date:   date,
+			Filter: images.FilterGTE,
+		},
+	}
+
+	allPages, err := images.List(client, listOpts).AllPages()
+	if err != nil {
+		t.Fatalf("Unable to retrieve all images: %v", err)
+	}
+
+	allImages, err := images.ExtractImages(allPages)
+	if err != nil {
+		t.Fatalf("Unable to extract images: %v", err)
+	}
+
+	for _, image := range allImages {
+		tools.PrintResource(t, image)
+		tools.PrintResource(t, image.Properties)
+	}
+
+	date = time.Date(2049, 1, 1, 1, 1, 1, 0, time.UTC)
+	listOpts = images.ListOpts{
+		Limit: 1,
+		CreatedAt: &images.ImageDateQuery{
+			Date:   date,
+			Filter: images.FilterGTE,
+		},
+	}
+
+	allPages, err = images.List(client, listOpts).AllPages()
+	if err != nil {
+		t.Fatalf("Unable to retrieve all images: %v", err)
+	}
+
+	allImages, err = images.ExtractImages(allPages)
+	if err != nil {
+		t.Fatalf("Unable to extract images: %v", err)
+	}
+
+	if len(allImages) > 0 {
+		t.Fatalf("Expected 0 images, got %d", len(allImages))
+	}
+}
+
+func TestImagesFilter(t *testing.T) {
```

Comment:
Is this testing the filter? It looks like it's just creating an image and listing all images.

Reply:
This test is using `ListOpts` to "filter" the image by certain attributes (lines 150-154).